app/script/backup/BackupRepository.js

```diff
@@ -84,45 +85,76 @@ z.backup.BackupRepository = class BackupRepository {
    * @returns {Promise<JSZip>} The promise that contains all the exported tables
    */
   generateHistory(progressCallback) {
-    const tables = this.backupService.getTables();
-    const metaData = this.createMetaDescription();
-    const tablesData = {};
     this.isCanceled = false;
+    const tables = this.backupService.getTables();
 
-    const loadDataPromises = tables.map(table => {
-      return this.backupService.exportTable(table, (tableName, rows) => {
-        if (this.isCanceled) {
-          throw new z.backup.CancelError();
-        }
-        progressCallback(rows.length);
+    return Promise.resolve()
+      .then(() => this._exportHistoryConversations(tables, progressCallback))
+      .then(exportedData => this._exportHistoryEvents(tables, progressCallback, exportedData))
```

Comment:
Try to avoid handing down return values (by wrapping them in an object for example)